app/angular/src/client/preview/types.ts

```diff
@@ -28,4 +38,5 @@ export interface StoryFnAngularReturnType {
   moduleMetadata?: NgModuleMetadata;
   template?: string;
   styles?: string[];
+  renderMode?: RenderMode;
```

Comment:
As always, on the wording I am not sure and hesitant. If you have any other idea or this one is not good. do not hesitate 🙏

Reply:
I looked to add it in the parameters but it doesn't seem possible to modify it in the decorators

Reply:
Interesting approach cc @tmeasday

Reply:
Can you tell me about why it needs to be modified in decorators @ThibaudAV?

Reply:
hum I don't know if it needs to be modified in decorators.
I wonder more if this PR's solution to add the `renderMode` in the `StoryFnAngularReturnType` is the right solution🤷‍♂️ 

The global objective is to allow each story to define its own rendering mode when the default one does not work.
`renderMode` allows to ignore the `forceRender` ( [src](https://github.com/storybookjs/storybook/blob/next/app/angular/src/client/preview/render.ts#L17) ) and always render completely the story (and not only props)

Reply:
Yeah, I think I understand the idea. I'm not sure if it is a good idea or not but I do feel like it should be a parameter if it is a good idea 🤷

Reply:
Yeah I also prefer parameters for this @ThibaudAV

Reply:
but, with parameters there is a problematic behavior :/ 
(if my memory is good) :
From the moment when I display a story with this parameter, it remains present. Even when I switch back to a story that shouldn't have this parameter anymore 


maybe I can add a commit to use a parameter and let you test?

Reply:
but, with parameters there is a problematic behavior :/
I don't remember why 😞 

maybe I can add a commit to use a parameter and try again 🤔

Reply:
aaa `voila` I remember ^^ @shilman @tmeasday 

The Angular render engine for storybook is "optimized" (it doesn't always render completely). But in some cases it is necessary to manually tell it to do the full rendering because a change that it cannot detect is made. 

And these changes can be made in the `decorators` ex here: 
https://github.com/storybookjs/storybook/pull/14226/commits/5697ddad601d89ca8e3329515619f63dbada69a6#diff-85417477bba9e40f090386e05b7558277ce2c064645846251038639da4201cfcR81
If the language changes then the next rendering is "full" 

the feature that this unlocks is that it allows to modify a `Providers` (or ext... module) of angular by `controls` 
Ex: live language change, theme change, ext... 

but it doesn't seem to be possible to modify `parameters` (or story context) dinamically in decorators to configure the `renderMode` parameter


do you understand my explanation ? do you have any ideas ?

Reply:
@ThibaudAV I think when globals change it should trigger a full refresh anyway, so that shouldn't be necessary. Is that not happening for you?

Reply:
In this case it should change the `forceRender` already present no ? 

But I'm not sure. Not all globals changes are about Angular. And there would still be the same problem if the change comes from Args+Control ? 

I think it's more confortable to allow each story to skip the optimization if it's in one of the marginal cases where it doesn't work 🤔 what do you think?

Reply:
I think we need to document the desired behavior better and make sure that each framework implements it properly. We've also talked about making parameters modifiable, but I don't know when that's going to happen. I don't like adding random flags to the render results if we can avoid it.

Reply:
Ok I understand, 
I don't really see how to help to do that, not many frameworks use the current `forceRender` and I don't know if they have the same problems :/

But yes making an identical pattern for each framework seems better :)

Ok then I propose to delete this commit ?? I isolated it for this reason ^^

The changes of the previous commit allow to ignore the `forceRender` with a "surface" check on angular metadata. This may be sufficient in many cases but not in all cases (like the one in example where it is the function return value that changes)

Reply:
sorry i missed the proposal. i'm ok with deleting this commit!

Reply:
let's figure this out separately unless it's blocking you

Reply:
@shilman I deleted the commit

It doesn't block me ;) 

Part of the problem is still there. But it doesn't seem to affect many users.To see with the issues, if it will be necessary to allow users to manually disable the optimization 🤷‍♂️